### ISSUE TITLE
False positive invalid-overridden-method with abc.abstractproperty

### ISSUE THREAD

**Issue by vojtechtrefny** [PyCQA/pylint#3150]
### Steps to reproduce
```
import abc

class A():

    @abc.abstractproperty
    def prop(self):
        return

class B(A):

    @property
    def prop(self):
        return
```

### Current behavior

```
test.py:12:4: W0236: Method 'prop' was expected to be 'method', found it instead as 'property' (invalid-overridden-method)
```

### Expected behavior

No warning.

### pylint --version output

```
$ pylint-3 --version
pylint 2.4.1
astroid 2.3.0
Python 3.8.0b4 (default, Aug 30 2019, 00:00:00) 
[GCC 9.2.1 20190827 (Red Hat 9.2.1-1)]
```

**Comment by PCManticore** [comment]
Thank you, makes sense.

**Comment by scop** [comment]
Hit this myself too, but with a somewhat different (?) scenario involving a nested property, posting here just in case it's not the same:
```
$ cat t.py 
def my_property(func):
    @property
    def _wrapper(self):
        pass
    return _wrapper

class A:
    @property
    def foo():
        return True

class B(A):
    @my_property
    def foo():
        return False

$ pylint t.py | grep invalid-overridden-method
t.py:14:4: W0236: Method 'foo' was expected to be 'property', found it instead as 'method' (invalid-overridden-method)
```

**Comment by scop** [comment]
> Hit this myself too, but with a somewhat different (?) scenario involving a nested property, posting here just in case it's not the same:

This one's still open, re-reported as #4368
